api/python/t4/bucket.py

```diff
@@ -199,8 +200,47 @@ def delete(self, key):
         Raises:
             if delete fails
         """
+        if key[-1] == '/':
```

Comment:
Maybe raise an error if `key` is empty - e.g. `if not key or key.endswith('/'):`

Reply:
Good idea, but I think this is a separate error. I went ahead added a separate `if not key` check.